src/UI/PlayerSetupController.java

```diff
@@ -0,0 +1,208 @@
+package UI;
+
+import java.awt.Color;
+import java.util.ArrayList;
+
+import AI.AILibrary;
+import CommandingOfficers.Commander;
+import CommandingOfficers.CommanderInfo;
+import CommandingOfficers.CommanderLibrary;
+import Engine.Driver;
+import Engine.GameInstance;
+import Engine.IController;
+import Engine.MapController;
+import Engine.OptionSelector;
+import Terrain.MapMaster;
+import UI.InputHandler.InputAction;
+
+/**
+ * Controller for choosing COs and colors after the map has been chosen.
+ * Left/Right changes the CO selector with focus.
+ * Up/Down changes the CO selected for that slot.
+ * LS/RS changes the color of the selected CO?
+ */
+public class PlayerSetupController implements IController
+{
+  // This OptionSelector determines which player we have under the cursor.
+  private OptionSelector playerSelector;
+  private OptionSelector categorySelector;
+
+  private GameBuilder gameBuilder = null;
+  PlayerSetupInfo[] coSelectors;
+
+  public enum SelectionCategories { COMMANDER, COLOR_FACTION, TEAM, AI, START };
+
+  private IController subMenu;
+
+  public PlayerSetupController( GameBuilder builder )
+  {
+    // Once we hit go, we plug all the COs we chose into our gameBuilder.
+    gameBuilder = builder;
+
+    // Set up our row/col selectors.
+    int numCos = gameBuilder.mapInfo.getNumCos();
+    playerSelector = new OptionSelector(numCos);
+    categorySelector = new OptionSelector(SelectionCategories.values().length);
+    categorySelector.setSelectedOption(SelectionCategories.START.ordinal()); // Best case is that no changes are needed.
+
+    // Create objects to keep track of the selected options for each player.
+    coSelectors = new PlayerSetupInfo[numCos];
+
+    // Start by making default CO/color selections.
+    for(int co = 0; co < numCos; ++co)
+    {
+      // Set up our option selection framework
+      coSelectors[co] = new PlayerSetupInfo(numCos, co, CommanderLibrary.getCommanderList(), UIUtils.getCOColors(), UIUtils.getFactions(), AILibrary.getAIList());
+    }
+  }
+
+  @Override
+  public boolean handleInput(InputAction action)
+  {
+    boolean exitMenu = false;
+    if(null != subMenu)
+    {
+      boolean exitSub = subMenu.handleInput(action);
+      if(exitSub)
+      {
+        // Only apply the change from the sub-menu if it exited via ENTER.
+        if(action == InputAction.ENTER)
```

Comment:
Hah, you don't even need to reset those controls on BACK, since you check it here.